src/ray/protobuf/common.proto

```diff
@@ -528,6 +528,10 @@ enum TaskStatus {
   SCHEDULED = 2;
   // The task finished successfully.
   FINISHED = 3;
+  // The task is submitted to a worker process to execute.
+  SUBMITTED_TO_WORKER = 4;
+  // The task has failed to be scheduled and being rescheduled.
+  RESCHEDULED = 5;
```

Comment:
Q: Maybe it is unnecessary, and we can reuse SCHEDULED state here.

Reply:
lgtm